src/parser/tokenizer.rs

```diff
@@ -903,32 +893,24 @@ mod tests {
 
     #[test]
     fn regex_with_flags() {
-        let kind = Tokenizer::new("/^[a-z]+$/i").next_token().unwrap().kind;
-        if let TokenKind::Regex(r) = kind {
-            // There's not a function on the `regex::Regex` to check its flags directly.
-            assert!(r.is_match("ABC"));
-            assert!(!r.is_match("\nABC\n"));
-        } else {
-            panic!("Expected regex token")
-        };
-
-        let kind = Tokenizer::new("/^[a-z]+$/m").next_token().unwrap().kind;
-        if let TokenKind::Regex(r) = kind {
-            // There's not a function on the `regex::Regex` to check its flags directly.
-            assert!(!r.is_match("ABC"));
-            assert!(r.is_match("\nabc\n"));
-        } else {
-            panic!("Expected regex token")
-        };
-
-        let kind = Tokenizer::new("/^[a-z]+$/im").next_token().unwrap().kind;
-        if let TokenKind::Regex(r) = kind {
-            // There's not a function on the `regex::Regex` to check its flags directly.
-            assert!(r.is_match("ABC"));
-            assert!(r.is_match("\nABC\n"));
-        } else {
-            panic!("Expected regex token")
-        };
```

Comment:
I think we should leave this test in, since it's meant to cover the tokenization of the flag syntax in the jsonata parse step. (If you need to temporarily mark it `#[ignore]` that's okay. The new test here is just asserting on `regress` behavior so I don't think we really need that one.